app/assets/javascripts/headshot.js

```diff
@@ -0,0 +1,206 @@
+                eval(funcName);
+            }
+			return true;
+		}
+		return false; // no hook defined
+	},
+	
+	set_api_url: function(url) {
+		// set location of upload API script
+		this.api_url = url;
+	},
+	
+	set_swf_url: function(url) {
+		// set location of SWF movie (defaults to headshot.swf in cwd)
+		this.swf_url = url;
+	},
+	
+	get_html: function(width, height, params) {
+		// Return HTML for embedding headshot capture movie
+		// Specify pixel width and height (640x480, 320x240, etc.)
+		// Server width and height are optional, and default to movie width/height
+        server_width = params.server_width ? params.server_width : width;
+        server_height = params.server_height ? params.server_height : height;
+        session_key_name = params.session_key_name ? params.session_key_name : '';
+        csrf_token = params.csrf_token ? params.csrf_token : '';
+		
+		var html = '';
+		var flashvars = 'shutter_enabled=' + (this.shutter_sound ? 1 : 0) + 
+			'&shutter_url=' + escape(this.shutter_url) + 
+			'&width=' + width + 
+			'&height=' + height + 
+			'&server_width=' + server_width + 
+			'&server_height=' + server_height +
+            '&session_key_name=' + session_key_name +
+            '&csrf_token=' + csrf_token;
+		
+		if (this.ie) {
+			html += '<object classid="clsid:d27cdb6e-ae6d-11cf-96b8-444553540000" codebase="'+this.protocol+'://download.macromedia.com/pub/shockwave/cabs/flash/swflash.cab#version=9,0,0,0" width="'+width+'" height="'+height+'" id="headshot_movie" align="middle"><param name="allowScriptAccess" value="always" /><param name="allowFullScreen" value="false" /><param name="movie" value="'+this.swf_url+'" /><param name="loop" value="false" /><param name="menu" value="false" /><param name="quality" value="best" /><param name="bgcolor" value="#ffffff" /><param name="flashvars" value="'+flashvars+'"/></object>';
+		}
+		else {
+			html += '<embed id="headshot_movie" src="'+this.swf_url+'" loop="false" menu="false" quality="best" bgcolor="#ffffff" width="'+width+'" height="'+height+'" name="headshot_movie" align="middle" allowScriptAccess="always" allowFullScreen="false" type="application/x-shockwave-flash" pluginspage="http://www.macromedia.com/go/getflashplayer" flashvars="'+flashvars+'" />';
+		}
+		
+		this.loaded = false;
+		return html;
+	},
+	
+	get_movie: function() {
+		// get reference to movie object/embed in DOM
+		if (!this.loaded) return alert("ERROR: Movie is not loaded yet");
+		var movie = document.getElementById('headshot_movie');
+		if (!movie) alert("ERROR: Cannot locate movie 'headshot_movie' in DOM");
+		return movie;
+	},
+	
+	set_stealth: function(stealth) {
+		// set or disable stealth mode
+		this.stealth = stealth;
+	},
+	
+	snap: function(url, callback, stealth) {
+		// take snapshot and send to server
+		// specify fully-qualified URL to server API script
+		// and callback function (string or function object)
+		if (callback) this.set_hook('onComplete', callback);
```

Comment:
Expected '{' and instead saw 'this'.<br>Identifier 'set_hook' is not in camel case.